src/app.js

```diff
@@ -2,34 +2,99 @@
 //We've got some basic info about Karen's home
 //Debug the type of data provided
 //Return the types concatenated in a single variable
+function moreAboutHome(address,distanceFromTown,hasNeighbours){
+    let result = typeof address+ typeof distanceFromTown + typeof hasNeighbours;
+    return result;
+
+}
+
+
 
 //Progression 2:
 //Check if the data given is of the right type
 //parents = String, noOfSiblings = Number, isNuclearFamily = Boolean
 
+function moreAboutKaren(parents,noOfSiblings,isNuclearFamily){
+    return typeof parents=='string' && typeof noOfSiblings=='number' && typeof isNuclearFamily=='boolean';  
+
+}
+
+
 //Progression 3:
 //Lily is suspicious about Karen's new friend
 //Karen tells her friend's age and even writes it down
 //Check which one those is not a number (NaN) and return that value
 
+function doesFriendExist(ageInText,ageInNumber){
+    if(isNaN(ageInText)){
+        return ageInText
+    }
+    else if(isNaN(ageInNumber)){
+        return ageInNumber
+    }
+    else{
+        return 0;
+    }
+
+}
+
+
+
 //Progression 4:
 //Lily gave Karen x sweets
 //Karen ate y sweets herself
 //On her way to the river, she ate another z sweets every n meters travelled
 //Her friend divided the remaining sweets into 2 parts for each
 //How many sweets did her friend get to eat?
+function sweetTooth(totalNoOfSweets,sweetsConsumedByKaren,sweetsConsumedInNMeters,metersToTravel){
+    if(isNaN(totalNoOfSweets)||isNaN(sweetsConsumedByKaren)||isNaN(sweetsConsumedInNMeters)||isNaN(metersToTravel)||arguments.length==0){
+        return "No sweets for Karen's friend"
+    }
+    let availableSweets = totalNoOfSweets - (sweetsConsumedByKaren+(sweetsConsumedInNMeters*metersToTravel));
```

Comment:
Is this necessary to do divided by 2 in another line? Any specific Purpose of doing this.